test/lsp.py

```diff
@@ -375,10 +399,12 @@ def test_publish_diagnostics_warnings(self, solc: JsonRpcProcess) -> None:
         self.expect_equal(report['uri'], self.get_test_file_uri(TEST_NAME), "Correct file URI")
         diagnostics = report['diagnostics']
 
+        markers = self.getFileMarkers(TEST_NAME)
+
         self.expect_equal(len(diagnostics), 3, "3 diagnostic messages")
-        self.expect_diagnostic(diagnostics[0], code=6321, lineNo=13, startEndColumns=(44, 48))
-        self.expect_diagnostic(diagnostics[1], code=2072, lineNo= 7, startEndColumns=( 8, 19))
-        self.expect_diagnostic(diagnostics[2], code=2072, lineNo=15, startEndColumns=( 8, 20))
+        self.expect_diagnostic(diagnostics[0], code=6321, marker=markers["@unusedReturnVariable"])
+        self.expect_diagnostic(diagnostics[1], code=2072, marker=markers["@unusedVariable"])
+        self.expect_diagnostic(diagnostics[2], code=2072, marker=markers["@unusedContractVariable"])
```

Comment:
@Marenz these expect_diagnostic calls should not be needed anymore. Just like isoltest for all the other tests, if one wants to add a test, only such test files should be needed to be touched and not this lsp.py file. :)

Reply:
I mean, yeah, that is the idea. This PR is only the first step as I keep repeating.